### ISSUE TITLE
Append result `nextExpectedVersion` is last written revision and not the next

### ISSUE THREAD

**Issue by Eadword** [EventStore/EventStore-Client-NodeJS#116]
Sample code where this issue became apparent:
```ts
import { EventStoreDBClient, FORWARDS, jsonEvent, NO_STREAM, START } from '@eventstore/db-client'
import { v4 as uuidv4 } from 'uuid'

describe('Eventstore GRPC Client', () => {
  let client: EventStoreDBClient

  beforeAll(() => {
    client = new EventStoreDBClient(
      {
        endpoint: {
          address: process.env.ES_HOSTNAME ?? 'eventstore',
          port: parseInt(process.env.ES_GRPC_PORT ?? '2113'),
        },
      },
      { insecure: true },
      {
        username: process.env.ES_USERNAME!,
        password: process.env.ES_PASSWORD!,
      },
    )
  })

  it('can connect to event store', async () => {
    const id = uuidv4().replace(/\-/g, '')
    const stream = uuidv4().replace(/\-/g, '')
    const event = jsonEvent({
      type: 'context-client-test',
      data: {
        a: id,
      },
    })

    const appendResult = await client.appendToStream(stream, [event], {
      expectedRevision: NO_STREAM,
    })
    // THIS IS THE ERROR
    expect(appendResult.nextExpectedVersion).toEqual(1n)

    const events = await client.readStream(stream, 10, {
      fromRevision: START,
      direction: FORWARDS,
    })
    expect(events).toHaveLength(1)
    expect((events[0].event!.data as any).a).toBe(id)
  })
})

```

I would expect that after writing to an empty stream, the next expected version would be `1` or `2` depending on if it is `0` or `1` indexed, but it seems that I am actually getting `0` which seems to indicate that it is returning the revision that was written and not the next version.

Personally I prefer having the last revision written returned, but the name needs to be changed or at minimum have more documentation around it assuming this is in fact doing what I think it is doing.

(Tested with alpha 12)

**Comment by George-Payne** [comment]
Hi @Eadword,

I understand your confusion, but its not the "next version after this one" its "next value to use for expected revision" (i.e. the current revision of the stream.)

The confusion probably isn't helped that the naming is incorrect, it should be "nextExpectedRevision".

I've fixed the naming and added some documentation around it here: #120

**Comment by Eadword** [comment]
I find this to be a good compromise. Thank you for your quick work @George-Payne!

I would close this, but I will leave that to you in case you wish to maintain it for tracking purposes.

**Comment by George-Payne** [comment]
fixed by: #120
